src/rust/engine/client/src/main.rs

```diff
@@ -82,10 +85,12 @@ async fn execute(start: SystemTime) -> Result<i32, String> {
   })?;
   env_logger::init_from_env(env_logger::Env::new().filter_or("__PANTS_LEVEL__", level.as_ref()));
 
+  // Now that the logger has been set up, we can retroactively log any dropped env vars.
+  for name in dropped {
+    log::warn!("Environment variable with non-UTF-8 name or value ignored: {name}");
```

Comment:
It seems like it would have been easy to keep name and value (as OsString) and just log both here. It seems slightly mean to not log the value in the case where the value was the issue - which, IIUC, is the only actual known issue case so far.

Reply:
My concern with that was that the value may be unprintable in practice (e.g., it could be several K of binary data). We already know we're on thin ice trying to print it here. The name is far more likely to be short and (close to) printable. I suppose we could truncate it. But either way I think the name is what matters in practice for remediating the problem.